### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by rwgdrummer**

### CODE REVIEW

File: maskgen/serialization/probes.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    if probe.targetMaskFileName is not None:
+        item['targetMaskFileName'] = os.path.basename(probe.targetMaskFileName)
+        if copyFileDirectory is not None:
+            shutil.copy(probe.targetMaskFileName,
+                        os.path.join(copyFileDirectory, os.path.basename(probe.targetMaskFileName)))
+    if probe.donorMaskFileName is not None:
+        if copyFileDirectory is not None:
+            shutil.copy(probe.donorMaskFileName,
+                        os.path.join(copyFileDirectory, os.path.basename(probe.donorMaskFileName)))
+        item['donorMaskFileName'] = os.path.basename(probe.donorMaskFileName)
```
Comment (mag8224) [inline maskgen/serialization/probes.py]: Is it ok if there is a targetMaskFileName, but the donorMaskFileName is None? Shouldn't there always be both a target and donor?

File: maskgen/serialization/probes.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    targetsegment = []
+    donorsegment = []
+    if probe.targetVideoSegments is not None:
+        for segment in probe.targetVideoSegments:
+            targetsegment.append(serialize_segment(segment))
+    if probe.donorVideoSegments is not None:
```
Comment (mag8224) [inline maskgen/serialization/probes.py]: Same as the masks for the videoSegments.

File: maskgen/serialization/probes.py (hunk 1 above)
Comment (rwgdrummer) [reply maskgen/serialization/probes.py]: In most cases, donors do not exist.  Donors are applicable to pastesampled and paste splice type operations.  All other operations generate a target mask without a donor.

File: maskgen/serialization/probes.py (hunk 2 above)
Comment (rwgdrummer) [reply maskgen/serialization/probes.py]: cool.  Same answer.

**Pull request merged**